R/epidist.R

```diff
@@ -45,27 +45,32 @@
 #' )
 #'
 #' # example using onset to admission as the metric
-#' epidist(pathogen = "ebola", delay_dist = "onset_to_admission")
-epidist  <- function(pathogen, delay_dist, study = NULL) {
-
+#' epidist(pathogen = "ebola", delay_dist = "onset_to_death")
+epidist  <- function(
+    pathogen, 
+    delay_dist = c("incubation", 
+                   "onset_to_admission", 
+                   "onset_to_death", 
+                   "serial_interval",
+                   "generation_time"), 
+    study = NULL) {
+  
+  # read the data to get possible pathogen names
+  params <- utils::read.csv(system.file(
+    "extdata",
+    "parameters.csv",
+    package = "epiparameter",
+    mustWork = TRUE
+  ))
+  
+  # match pathogen names against data
   pathogen <- match.arg(
     arg = pathogen,
-    choices = c(
-      "adenovirus", "ebola", "human_CoV", "influenza_A_seasonal",
-      "influenza_B_seasonal", "influenza_H1N1p", "influenza_H7N9", "marburg",
-      "measles", "MERS_CoV", "parainfluenza", "rhinovirus", "RSV", "SARS_CoV",
-      "SARS_CoV_2_wildtype", "monkeypox"
-    ),
+    choices = unique(params$pathogen_id),
     several.ok = FALSE
   )
 
-  delay_dist <- match.arg(
-    arg = delay_dist,
-    choices = c(
-      "incubation", "onset_to_admission", "onset_to_death", "serial_interval"
-    ),
-    several.ok = FALSE
-  )
+  delay_dist <- match.arg(arg = delay_dist, several.ok = FALSE)
 
   # Extract pathogen and parameter type
   pick_path <- utils::read.csv(file = system.file(
```

Comment:
You read this file just before (and saved it as `params`). It would be much better to read it just once, save it with a more generic name, and then re-use it here.

This will become more important as the database grows.

Reply:
Resolved in 8427837b95fc718f7eed8f96283ee76bc925f917.